core/src/com/ericc/the/game/mapGenerator/Generator.java

```diff
@@ -0,0 +1,147 @@
+package com.ericc.the.game.mapGenerator;
+
+import java.util.Random;
+
+public class Generator {
+    private int width, height;
+    private Tile[][] map;
+    private Random random;
+    private int maximalRoomSize;
+
+    Generator(int width, int height, int maximalRoomSize) {
+        this.width = width;
+        this.height = height;
+        this.maximalRoomSize = maximalRoomSize;
+
+        random = new Random();
+        map = new Tile[width][height];
+
+        for (int i = 0; i < width; ++i) {
+            for (int j = 0; j < height; ++j) {
+                map[i][j] = Tile.generateNotAvailableTile();
+            }
+        }
+    }
+
+    /**
+     * fills the given room (it randomizes a little bit the given arguments),
```

Comment:
it randomizes a little bit the given arguments -> it randomizes the given arguments a little bit